app/src/androidTest/java/org/simple/clinic/DatabaseMigrationAndroidTest.kt

```diff
@@ -3352,6 +3355,85 @@ class DatabaseMigrationAndroidTest {
         "deletedAt")
     )
   }
+
+  @Test
+  fun migration_from_48_to_49_create_only_one_encounter_for_per_day_but_add_encounterUuid_to_all_BPs() {
+    //given
+    val tableName = "BloodPressureMeasurement"
+
+    val uuidForFirstBP = "3eb96138-5270-419c-bcdf-86db95d0bc8b"
+    val uuidForSecondBP = "a2455b0c-cf14-46c9-9f73-6e18972982b0"
+    val facilityUuid = "0086ba86-314c-49d9-8822-2563e53bed92"
+    val patientUuid = "c4233170-feb2-3d34-8115-98fef5a68837"
+    val recordedAtForFirstBP = "2018-07-03T00:00:00Z"
+    val recordedAtForSecondBP = "2018-07-03T01:30:00Z"
+    val createdAtForFirstBP = "2018-07-03T00:00:00Z"
+    val createdAtForSecondBP = "2018-07-03T01:30:00Z"
+
+    val db_47 = helper.createDatabase(version = 48)
+    db_47.insert(tableName, mapOf(
+        "uuid" to uuidForFirstBP,
+        "systolic" to "120",
+        "diastolic" to "90",
+        "syncStatus" to "DONE",
+        "userUuid" to "d689285f-5c92-4735-80ef-c0e360ad3f1d",
+        "facilityUuid" to facilityUuid,
+        "patientUuid" to patientUuid,
+        "createdAt" to createdAtForFirstBP,
+        "updatedAt" to createdAtForFirstBP,
+        "deletedAt" to "null",
+        "recordedAt" to recordedAtForFirstBP
+    ))
+
+    db_47.insert(tableName, mapOf(
+        "uuid" to uuidForSecondBP,
+        "systolic" to "130",
+        "diastolic" to "60",
+        "syncStatus" to "DONE",
+        "userUuid" to "d689285f-5c92-4735-80ef-c0e360ad3f1d",
+        "facilityUuid" to facilityUuid,
+        "patientUuid" to patientUuid,
+        "createdAt" to createdAtForSecondBP,
+        "updatedAt" to createdAtForSecondBP,
+        "deletedAt" to "null",
+        "recordedAt" to recordedAtForSecondBP
+    ))
+
+    //when
+    val db_49 = helper.migrateTo(version = 49)
+
+    //then
+    val expectedEncounteredOn = Instant.parse(recordedAtForFirstBP).toLocalDateAtZone(ZoneOffset.UTC).toString()
+    val expectedEncounterId = createUuid5(facilityUuid + patientUuid + expectedEncounteredOn).toString()
```

Comment:
`createUuid5(facilityUuid + patientUuid + expectedEncounteredOn)` this method signature reveals too much information and does not reveal intent, can we change the signature to `createUuid5(facilityUuid, patientUuid, expectedEncounteredOn)`?

Reply:
I have added a separate method which is responsible for generating UUIDs for `Encounter`.
PTAL. :)